invenio_communities/assets/semantic-ui/js/invenio_communities/members/searchBarWithFilters.js

```diff
@@ -0,0 +1,244 @@
+  currentQueryState: PropTypes.func.isRequired,
+  sort: PropTypes.string.isRequired,
+};
+
+export class SearchBarWithFilters extends Component {
+  render() {
+    const {
+      filters,
+      sortOptions,
+      currentQueryState,
+      updateQueryState,
+      searchBarPlaceholder,
+      customCmp,
+      ...uiProps
+    } = this.props;
+    const currentFilters = currentQueryState.filters;
+    const currentSort = currentQueryState.sortBy;
+    return (
+      <Grid.Row>
+        <Grid.Column
+          className="searchbar-with-filters"
+          computer={16}
+          tablet={16}
+        >
+          <SearchBar
+            uiProps={{ className: "searchbar-with-filters" }}
+            placeholder={searchBarPlaceholder}
+          />
+          <div>
+            {Object.entries(filters).map((filter) => (
+              <DropdownFilter
+                className="ml-20"
+                filterKey={filter[0]}
+                filterLabel={filter[1].label}
+                filterValues={filter[1].buckets}
+                currentQueryState={currentQueryState}
+                updateQueryState={updateQueryState}
+                {...uiProps}
+              />
+            ))}
+            {sortOptions && (
+              <DropdownSort
+                className="ml-20"
+                filterKey="sort"
+                filterLabel={i18next.t("Sort")}
+                filterValues={sortOptions}
+                currentQueryState={currentQueryState}
+                updateQueryState={updateQueryState}
+                {...uiProps}
+              />
+            )}
+          </div>
+          {customCmp && customCmp}
+        </Grid.Column>
+
+        {currentFilters.map((filter) => {
+          return (
+            <FilterLabel
+              filter={filter}
+              currentQueryState={currentQueryState}
+              updateQueryState={updateQueryState}
+            />
+          );
+        })}
+        {currentSort && (
+          <SortLabel
+            sort={currentSort}
+            currentQueryState={currentQueryState}
+            updateQueryState={updateQueryState}
+          />
+        )}
+        {currentFilters.length > 0 && (
+          <Label
+            as="a"
+            onClick={() => {
+              currentQueryState.filters = [];
+              currentQueryState.sortBy = undefined;
+              updateQueryState(currentQueryState);
+            }}
+            color="red"
+            className="ml-15 mt-15"
+          >
+            {i18next.t("Clear all filters")}
+          </Label>
+        )}
+      </Grid.Row>
+    );
+  }
+}
+
+SearchBarWithFilters.propTypes = {
+  updateQueryState: PropTypes.object.isRequired,
+  currentQueryState: PropTypes.func.isRequired,
+  sortOptions: PropTypes.array.isRequired,
+  filters: PropTypes.object.isRequired,
+  searchBarPlaceholder: PropTypes.string,
+  customCmp: PropTypes.node,
+};
+
+SearchBarWithFilters.defaultProps = {
+  searchBarPlaceholder: "Search ...",
```

Comment:
Should the placeholder be translated somehow?

Reply:
Yes, thanks! Will be addresed in the next PR.

Reply:
addressed